oemof_b3/tools/data_processing.py

```diff
@@ -1,7 +1,761 @@
+
+    required_header : list
+        list of strings with required positions in the header
+
+    """
+    if data_type == "scalars":
+        # Name of each column in scalars
+        header = [
+            "id_scal",
+            "scenario",
+            "name",
+            "var_name",
+            "carrier",
+            "region",
+            "tech",
+            "type",
+            "var_value",
+            "var_unit",
+            "reference",
+            "comment",
+        ]
+        # Names of optional columns in scalars
+        optional_header = ["id_scal", "var_unit", "reference", "comment"]
+
+    elif data_type == "timeseries":
+        # Names of all columns in a stacked time series
+        header = [
+            "id_ts",
+            "region",
+            "var_name",
+            "timeindex_start",
+            "timeindex_stop",
+            "timeindex_resolution",
+            "series",
+            "var_unit",
+            "source",
+            "comment",
+        ]
+
+        # Names of optional columns in a stacked time series
+        optional_header = [
+            "id_ts",
+            # "region",
+            "var_unit",
+            "source",
+            "comment",
+        ]
+    else:
+        raise ValueError(
+            f"{data_type} is not a valid option of a description of the DataFrame type. "
+            f"Please choose between 'scalars' and 'timeseries'."
+        )
+
+    # Names of required columns in scalars
+    required_header = header.copy()
+    for optional in optional_header:
+        required_header.remove(optional)
+
+    return header, optional_header, required_header
+
+
+def load_scalars(path):
+    """
+    This function loads scalars from a csv file
+
+    Parameters
+    ----------
+    path : str
+        path of input file of csv format
+    Returns
+    -------
+    df : pd.DataFrame
+        DataFrame with loaded scalars
+
+    """
+    # Get header of scalars
+    scalars_header = get_optional_required_header("scalars")
+    header = scalars_header[0]
+    optional_header = scalars_header[1]
+    required_header = scalars_header[2]
+
+    # Get file name
+    filename = os.path.splitext(path)[0]
+
+    # Read data
+    df = pd.read_csv(path)
+
+    # Save header of DataFrame to variable
+    df_header = list(df.columns)
+
+    # Check whether required columns are missing in the DataFrame
+    missing_required = []
+    for required in required_header:
+        if required not in df_header:
+            # Add required columns, that are missing, to a list
+            missing_required.append(required)
+
+    # Interrupt if required columns are missing and print all affected columns
+    if len(missing_required) > 0:
+        raise KeyError(
+            f"The data in {filename} is missing the required column(s): {missing_required}"
+        )
+
+    # Check whether optional columns are missing
+    for optional in optional_header:
+        if optional not in df_header:
```

Comment:
Because it does not appear in the results, neither in the scalar nor in the time series. So I wrote the script in a way that the id is variable and this column (which is just a numbering of the DataFrame, isn't it?) is added in case it is missing.

Reply:
I made a suggestion above how to handle this.

Reply:
Unfortunately, I do not understand how this makes a difference to my variant. In both cases `id_scal` is added when it does not exist. If I set `id_scal` as index, I have to adjust the tests, because then it doesn't show up there as column anymore. Also saving the DataFrame I have to unindex `id_scal` so that this column is saved with the respective header don't I? Hence I see a lot of extra work with the index variant.